packages/actor-rdf-resolve-quad-pattern-sparql-json/lib/ActorRdfResolveQuadPatternSparqlJson.ts

```diff
@@ -26,6 +26,51 @@ export class ActorRdfResolveQuadPatternSparqlJson
     super(args);
   }
 
+  /**
+   * Replace all blank nodes in a pattern with variables.
+   * If the pattern contains no blank nodes the original pattern gets returned.
+   * @param {RDF.Quad} pattern A quad pattern.
+   * @return {RDF.Quad} A quad pattern with no blank nodes.
+   */
+  public static replaceBlankNodes(pattern: RDF.Quad): RDF.Quad {
+    const variableNames: string[] = getVariables(getTerms(pattern)).map((v) => v.value);
+    const result: { [id: string]: RDF.Term } = {};
+    // track the names the blank nodes get mapped to (required if the name has to change)
+    const blankMap: { [id: string]: string } = {};
+    let changed = false;
+
+    // for every position, convert to a variable if there is a blank node
+    for (const pos of QUAD_TERM_NAMES) {
```

Comment:
I would suggest simplifying this loop using [`mapQuadTerms`](https://www.npmjs.com/package/rdf-terms#map-quad-terms), this would also allow us to get rid of the `<any>` cast on the next line.

Reply:
actually didn't know that existed :D

Reply:
As far as I can see that function only exists in the documentation but not actually in the library

Reply:
README seems to be incorrect. The actual name is `mapTerms` (will fix the README right away!)